airflow/www/security.py

```diff
@@ -233,8 +233,8 @@ def get_all_permissions_views(self):
         """
         perms_views = set()
         for role in self.get_user_roles():
-            for perm_view in role.permissions:
-                perms_views.add((perm_view.permission.name, perm_view.view_menu.name))
+            perms_views = perms_views | {(perm_view.permission.name, perm_view.view_menu.name)
+                                         for perm_view in role.permissions}
```

Comment:
```python
def fun1(data):
    a=set()
    for d in data:
        a.add(d)
    return a

def fun2(data):
    a=set()
    a = a | {d for d in data}
    return a

if __name__ == '__main__':
    import timeit
    print(fun1(list(range(100)) + list(range(100))) == fun2(list(range(100)) + list(range(100))))
    print(timeit.timeit("fun1(list(range(300)) + list(range(300)))", setup="from __main__ import fun1"))
    print(timeit.timeit("fun2(list(range(300)) + list(range(300)))", setup="from __main__ import fun2"))
```
Result:
True
52.7378087699999
27.333367159999398

Reply:
hey @XD-DENG , do you know where does the performance improvement happen ?

Reply:
Hi @feng-tao , please correct me if I'm wrong: the difference is mainly between adding "one-by-one" or "adding in bulk".

I just realise `a.update({d for d in data})` may be more neat than `a = a | {d for d in data}`, while they're equivalent. Will make this change and push to trigger CI again (CI failed because of the RedisPy issue).

Reply:
Updated benchmarking:

```python
def fun1(data):
    a=set()
    for d in data:
        a.add(d)
    return a

def fun2(data):
    a=set()
    a = a | {d for d in data}
    return a

def fun3(data):
    a=set()
    a.update({d for d in data})
    return a

if __name__ == '__main__':
    import timeit
    print(fun1(list(range(100)) + list(range(100))) == fun2(list(range(100)) + list(range(100))))
    print(fun2(list(range(100)) + list(range(100))) == fun3(list(range(100)) + list(range(100))))
    print(timeit.timeit("fun1(list(range(300)) + list(range(300)))", setup="from __main__ import fun1"))
    print(timeit.timeit("fun2(list(range(300)) + list(range(300)))", setup="from __main__ import fun2"))
    print(timeit.timeit("fun3(list(range(300)) + list(range(300)))", setup="from __main__ import fun3"))
```

Result:
True
True
51.447706788999994
26.681957754999985
26.59145899200007

Reply:
Rebased and made the change mentioned above.

Please let me know if anything looks incorrect. Thanks!